.github/workflows/lint.yml

```
@@ -0,0 +1,95 @@
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff black
+
+      - name: Run ruff --fix
+        run: ruff check --fix . || true
+
+      - name: Run black
+        run: black . || true
+
+      - name: Commit & push fixes
+        run: |
```

actionlint (1.7.12)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)